core/CronArchive/QueueConsumer.php

```diff
@@ -541,4 +548,10 @@ public function usableArchiveExists(array $invalidatedArchive)
         $idArchive = $archiveIdAndVisits[0];
         return !empty($idArchive);
     }
+
+    private function isSiteExists($idSite)
+    {
+        $site = API::getInstance()->getSiteFromId($idSite);
```

Comment:
@diosmosis can we use the `Site` class here so it uses cached data? It be fine if only the next cronjob would notice the site was actually removed and that this one would basically "fail" when the site was deleted. Just to save some IO etc. The next cronjob could then make sure to remove the invalidations for that site as then the in-memory cache would be correct

Reply:
@tsteur can't archiving runs take hours and hours? This would mean the in memory cache would be inaccurate for the entire run, correct?

Reply:
It can but it should be fine as usually also every hour a different job would start and mark them then as deleted basically

Reply:
@tsteur if archive processes never process the same site, then will this still happen? I guess we could do it all at once at the start of a core:archive process, but then for many hours long processes a site could be deleted in the middle of one, and we'd continue archiving...

Reply:
True. Each site would be picked only once so a cache wouldn't be too important and in general the extra code for it might not be too important. I think at some point the archiving would simply fail for the site if it's removed in the middle of archiving. This should be fine as it would be a rare case and it would work the next time. I'm really just wanting to make sure we're cleaning up archive_invalidations table when a site is deleted. This could be done like you mentioned in the deleteSite event or so for now.

Reply:
Will add it to a daily task, seems easier to handle all cases that way (ie, look for deleted sites and delete invalidations w/ those idsites).

Reply:
@diosmosis generally might be could to also do it on deleteSite event if any possible using the same code. This way it be basically instantly applied and we might prevent getting some failed archive emails because of it.